test/Microsoft.ML.Tests/TrainerEstimators/MatrixFactorizationTests.cs

```diff
@@ -393,23 +404,22 @@ public void OneClassMatrixFactorizationInMemoryDataZeroBaseIndex()
             // Create a matrix factorization trainer which may consume "Value" as the training label, "MatrixColumnIndex" as the
             // matrix's column index, and "MatrixRowIndex" as the matrix's row index.
             var mlContext = new MLContext(seed: 1, conc: 1);
-            var pipeline = mlContext.Recommendation().Trainers.MatrixFactorization(
-                nameof(OneClassMatrixElementZeroBased.MatrixColumnIndex),
-                nameof(OneClassMatrixElementZeroBased.MatrixRowIndex),
-                nameof(OneClassMatrixElementZeroBased.Value),
-                advancedSettings: s =>
-                {
-                    s.LossFunction = MatrixFactorizationTrainer.LossFunctionType.SquareLossOneClass;
-                    s.NumIterations = 100;
-                    s.NumThreads = 1; // To eliminate randomness, # of threads must be 1.
-                    // Let's test non-default regularization coefficient.
-                    s.Lambda = 0.025;
-                    s.K = 16;
-                    // Importance coefficient of loss function over matrix elements not specified in the input matrix.
-                    s.Alpha = 0.01;
-                    // Desired value for matrix elements not specified in the input matrix.
-                    s.C = 0.15;
-                });
+
+            var options = new MatrixFactorizationTrainer.Options
+            {
+                MatrixColumnIndexColumnName = nameof(MatrixElement.MatrixColumnIndex),
+                MatrixRowIndexColumnName = nameof(MatrixElement.MatrixRowIndex),
+                LabelColumnName = nameof(MatrixElement.Value),
+                LossFunction = MatrixFactorizationTrainer.LossFunctionType.SquareLossOneClass,
+                NumIterations = 100,
+                NumThreads = 1,
```

Comment:
Missing my old comment.

Reply:
fixed

---
In reply to: [250022861](https://github.com/dotnet/machinelearning/pull/2204#discussion_r250022861) [](ancestors = 250022861)